File: Yunhyunjo/Codingtest/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    brown = (brown - 4) / 2;
    int x;
    if(brown % 2 == 0) x = brown / 2;
    else x = brown / 2 + 1;
    int y = brown - x;
    while(x*y != yellow){
        x++;
        y--;
    }
    answer.push_back(x+2);
    answer.push_back(y+2);
    return answer;
}

File: gut27/Codingtest/카펫.py
def solution(brown, yellow):
    answer = []
    num_sum = brown + yellow
    arr = []
    for y in range(3, (num_sum//3)+1):
        if num_sum % y ==0:
            x = num_sum//y
            if (x-2)*(y-2) == yellow:
                answer = [x,y]
                return answer
